frontend/src/metabase/search/containers/SearchApp.unit.spec.tsx

```diff
@@ -0,0 +1,215 @@
+import userEvent from "@testing-library/user-event";
+import { renderWithProviders, screen, waitFor } from "__support__/ui";
+import SearchApp from "metabase/search/containers/SearchApp";
+import { Route } from "metabase/hoc/Title";
+import {
+  setupDatabasesEndpoints,
+  setupSearchEndpoints,
+  setupTableEndpoints,
+} from "__support__/server-mocks";
+import {
+  createMockDatabase,
+  createMockSearchResult,
+  createMockTable,
+} from "metabase-types/api/mocks";
+import { SearchResult } from "metabase-types/api";
+
+import { SearchFilters } from "metabase/search/types";
+
+// Mock PAGE_SIZE so we don't have to generate a ton of elements for the pagination test
+jest.mock("metabase/search/containers/constants", () => ({
+  PAGE_SIZE: 3,
+}));
+
+const ALL_RESULTS_SIDEBAR_NAME = "All results";
+
+const SIDEBAR_NAMES: Record<string, string> = {
+  collection: "Collections",
+  dashboard: "Dashboards",
+  database: "Databases",
+  dataset: "Models",
+  metric: "Metrics",
+  pulse: "Pulses",
+  segment: "Segments",
+  table: "Raw Tables",
+  card: "Questions",
+};
+
+const TEST_ITEMS: Partial<SearchResult>[] = [
+  { name: "Test Card", model: "card" },
+  { name: "Test Collection", model: "collection" },
+  { name: "Test Dashboard", model: "dashboard" },
+  { name: "Test Database", model: "database" },
+  { name: "Test Dataset", model: "dataset" },
+  { name: "Test Table", model: "table" },
+  { name: "Test Pulse", model: "pulse" },
+  { name: "Test Segment", model: "segment" },
+  { name: "Test Metric", model: "metric" },
+];
+
+const TEST_SEARCH_RESULTS: SearchResult[] = TEST_ITEMS.map((metadata, index) =>
+  createMockSearchResult({ ...metadata, id: index + 1 }),
+);
+
+const TEST_DATABASE = createMockDatabase();
+const TEST_TABLE = createMockTable();
+
+const setup = async ({
+  searchText,
+  searchFilters = {},
+  searchItems = TEST_SEARCH_RESULTS,
+}: {
+  searchText: string;
+  searchFilters?: SearchFilters;
+  searchItems?: SearchResult[];
+}) => {
+  setupDatabasesEndpoints([TEST_DATABASE]);
+  setupSearchEndpoints(searchItems);
+  setupTableEndpoints(TEST_TABLE);
+
+  // for testing the hydration of search text and filters on page load
+  const params = {
+    ...searchFilters,
+    q: searchText,
+  };
+
+  const searchParams = new URLSearchParams(
+    params as unknown as Record<string, string>,
+  ).toString();
+
+  const initialRoute = searchParams ? `/search?${searchParams}` : `/search`;
+
+  const { history } = renderWithProviders(
+    <Route path="search" component={SearchApp} />,
+    {
+      withRouter: true,
+      initialRoute,
+    },
+  );
+
+  await waitFor(() => {
+    expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument();
+  });
+
+  return {
+    history,
```

Comment:
Have you tried using `checkNotNull`? It should help get rid of any `history?.xxx` call out there since we know `history` will not be null in this case.

Reply:
Huh, I thought I changed this already since we talked about this before... I'll change it again